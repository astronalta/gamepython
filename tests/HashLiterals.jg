# jgtest exec on

Intf < Interface {
    x() String {
    } 
}

One < Object {
    x() String { 
        ret "Hello"
    }
}

Two < Object {
    x() String {
        ret "Goodbye"
    }
}

HashLike < Object {
    @insert(key String, val String) {
        Io::println("#{key} => #{val}")
    }
}

Key < Object {

    val Int

    @init(val Int) {
        self.val = val
    }
}

main() Int {
    hash = { 'a': 'one', 'b': 'two' }
    Io::println("#{'a'} => #{hash['a']}")
    Io::println("#{'b'} => #{hash['b']}")

    hash = Hash[String,String]()
    for pair in hash {
        Io::println("#{pair.first} => #{pair.second}")
    }

    hash2 Hash[String,Intf] = { 'a': One(), 'b': Two() }
    Io::println("#{'a'} => #{hash2['a'].x()}")
    Io::println("#{'b'} => #{hash2['b'].x()}")

    hash3 HashLike = { '1': 'Spock', '2': 'Kirk', '3': 'Picard' }
    
    k1 = Key(1)
    k2 = Key(2)
    hash9 = { k1: 'Bilbo', k2: 'Frodo' } 
    Io::println("#{1} => #{hash9[k1]}")
    Io::println("#{2} => #{hash9[k2]}")
    ret 0
}
