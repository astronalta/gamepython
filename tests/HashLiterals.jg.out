a => one
b => two
a => Hello
b => Goodbye
1 => Spock
2 => Kirk
3 => Picard
1 => Bilbo
2 => Frodo
